lib/Sema/CSBindings.cpp

```diff
@@ -366,11 +366,57 @@ void PotentialBindings::inferTransitiveBindings(
   }
 }
 
+// If potential binding type variable or any of it's adjacents represents the
+// argument in an argument conversion constraint located directly on an
+// autoclosure parameter.
+static bool
+isClosureInAutoClosureArgumentConversion(PotentialBindings &bindings) {
+  auto &CG = bindings.CS.getConstraintGraph();
+  auto isClosureVarInAutoClosureContext = [&](TypeVariableType *typeVar) {
+    if (!typeVar->getImpl().isClosureType())
+      return false;
+
+    auto constraints = CG.gatherConstraints(
+        typeVar, ConstraintGraph::GatheringKind::EquivalenceClass);
```

Comment:
This is very expensive, is there s way to avoid this?

Reply:
Hey @xedin :) 
Yeah, I give some thought on that, but couldn't come up with anything else since this have to be stateless. All other options I could think involves storing some kind of state while inferring the bindings...

Reply:
Let me think a bit more about this. Ping you when I have some question or solution =]

Reply:
We'd need to figure out a way here, maybe store some more information in the solution or argument information caches we already have stored in the constraint system...

Reply:
Right, I'll look into that. Thanks :)

Reply:
I've done some changes, let me know what you think :)